### PR TITLE
Added cookies_ingest attribute to torrent providers. Should fix the TD captcha.

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

Users will now be able to configure cookies for providers with the cookies_ingest attribute.
This is now only enabled for torrentday. To fix the captcha issue.

### CODE REVIEW

**Comment by fernandog** [comment]
@p0psicles nice job! working for me

File: sickrage/providers/torrent/TorrentProvider.py [hunk 1]
```diff
@@ -192,3 +193,12 @@ def _get_hash(self, item):
             hash = None
 
         return hash
+
+    @staticmethod
+    def split_cookies(cookie):
+        try:
+            return {k.strip(): v.strip() for k, v in [cookie.split('=') for cookie in cookie.split(';')]}
+        except Exception:
+            logger.log('Could not parse cookie! Check your input!'.format
```
Comment (p0psicles) [inline sickrage/providers/torrent/TorrentProvider.py]: Oops

**Comment by p0psicles** [comment]
I've decided to rename generalize the cookie handling. I'm going to:
* Rename the provider.cookies attribute for the rss_torrent to provider.rss_cookies
* Use the attribute provider.cookies as a generic attribute, usuable by all providers
* Move the code to add the cookie to a method in GenericProvider (making it possible to only have to call self.add_cookies_from_ui() from the provider)

This may look allot of work, but it's not really. Need to only do some UI/js changes, and this is done.

**Comment by p0psicles** [comment]
This needs to be tested some more. But now also for rss_torrent regresssion.
@fernandog @duramato ?

File: sickbeard/providers/torrentday.py [hunk 2]
```diff
@@ -167,27 +168,20 @@ def login(self):
                 'submit.y': 0,
             }
 
-            response = self.get_url(self.urls['login'], post_data=login_params, returns='text')
-            if not response:
+            response = self.get_url(self.urls['login'], post_data=login_params, returns='response')
+            if response.status_code not in [200]:
```
Comment (medariox) [inline sickbeard/providers/torrentday.py]: Why not
```
if response.status_code != 200:
```
?

Comment (p0psicles) [inline sickbeard/providers/torrentday.py]: Ooh yeah I started with testing for multiple response codes

Comment (p0psicles) [inline sickbeard/providers/torrentday.py]: I'm doing some more rewrites.  Replaced the tuples with dicts, and fixed some issues when you input bad formatted cookies.

**Comment by medariox** [comment]
Approved

**Pull request merged**